### PR TITLE
Allow for reading TOML files from stdin.

### LINKED ISSUE #238: Allow for reading from stdin
I'd like to configure pyproject-fmt as [formatter in Zed](https://zed.dev/docs/configuring-zed#formatter). Formatters in Zed need to be able to read from stdin and write to stdout. It seems pyproject-fmt can do the latter, but not the former. 

Is this something you would consider adding? I can provide a PR if you'd like.

I was thinking of adding this as flag under run mode:

```  
run mode:
   -s, --stdout          print the formatted TOML to the stdout (default: False)
   -i, --stdin           read the TOML to format from the stdin (default: False)
```

### PR DESCRIPTION
**Pull request by fniessink**
When the input filename is '-', read the toml from stdin and output to stdout.

Closes #238.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,9 @@
 *.egg-info/
+.coverage
```
Comment (gaborbernat) [inline .gitignore:2]: Revert changes to this files, and use tox to manage your development environments.

File: src/pyproject_fmt/cli.py [hunk 2]
```diff
@@ -151,6 +149,50 @@ def _build_cli() -> ArgumentParser:
     return parser
 
 
+def _settings(opt: PyProjectFmtNamespace, pyproject_toml: dict[str, Any]) -> Settings:
```
Comment (gaborbernat) [inline src/pyproject_fmt/cli.py:152]: Please move it back to inline, moving it out part of this PR makes the PR hard to review.

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: src/pyproject_fmt/cli.py [hunk 3]
```diff
@@ -60,6 +56,8 @@ def pyproject_toml_path_creator(argument: str) -> Path:
     :param argument: the string argument passed in
     :return: the pyproject.toml path
     """
+    if argument == "-":
```
Comment (gaborbernat) [inline src/pyproject_fmt/cli.py:59]: Path("-") makes no sense, perhaps this should return None instead for this cases.

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: src/pyproject_fmt/cli.py (hunk 3 above)
Comment (fniessink) [reply src/pyproject_fmt/cli.py:59]: Replaced `Path("-")` with `None`.

File: tests/test_cli.py [hunk 4]
```diff
@@ -65,6 +66,15 @@ def test_cli_inputs_ok(tmp_path: Path) -> None:
     assert len(result) == 3
 
 
+def test_cli_pyproject_toml_stdin(monkeypatch: pytest.MonkeyPatch) -> None:
+    monkeypatch.setattr("sys.stdin", io.StringIO(""))
```
Comment (gaborbernat) [inline tests/test_cli.py:70]: Use `pytest-mock` instead.

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: tests/test_cli.py (hunk 4 above)
Comment (fniessink) [reply tests/test_cli.py]: Done.

**Review by gaborbernat — APPROVED** [review]

**Review by fniessink — COMMENTED** [review]
How would you like me to deal with the typing error in `__main__.py`?

In `_handle_one`, after the `if config.stdout` block, `config.pyproject_toml` is never `None`, but the type checker can't detect that:

```python
def _handle_one(config: Config) -> bool:
    formatted = format_toml(config.toml, config.settings)
    before = config.toml
    changed = before != formatted
    if config.stdout:  # stdout just prints new format to stdout
        print(formatted, end="")  # noqa: T201
        return changed

    if before != formatted and not config.check:
        config.pyproject_toml.write_text(formatted, encoding="utf-8")
```

I see a few options: 
- casting to `Path`
- changing the `if config.stdout` to `if config.pyproject_toml is None or config.stdout`

There are probably other options as well.

**Comment by gaborbernat** [comment]
either is fine

**Review by gaborbernat — APPROVED** [review]

**Comment by fniessink** [comment]
Now with a signed commit.

**Review by gaborbernat — APPROVED** [review]

**Pull request merged**